internal/rancher/client.go

```diff
@@ -29,20 +29,22 @@ func NewClient() *Client {
 }
 
 type Client struct {
-	c           *http.Client
-	cachedToken KubeconfigToken
-	mux         sync.Mutex
-	password    string
-	timeout     time.Duration
-	url         string
-	username    string
+	c               *http.Client
+	cachedToken     KubeconfigToken
+	mux             sync.Mutex
+	password        string
+	shortExpiration int // seconds for the expiration
+	timeout         time.Duration
+	url             string
+	username        string
 }
 
 func (c *Client) Token(ctx context.Context) (string, error) {
 	c.mux.Lock()
 	defer c.mux.Unlock()
 
-	if time.Now().In(time.UTC).After(c.cachedToken.ExpiresAt) || c.cachedToken.Token == "" {
+	tokenExpired := int(time.Since(c.cachedToken.Created.In(time.UTC)).Seconds()) > c.shortExpiration
+	if time.Now().In(time.UTC).After(c.cachedToken.ExpiresAt) || c.cachedToken.Token == "" || tokenExpired {
```

Comment:
Maybe these two lines could benefit from an unexported function, converting it to look like
```go
if c.expired() { // or c.tokenExpired()
   // logic here stays the same
}

Reply:
Done! Thanks, I like this solution a lot better, feels cleaner.